### PR TITLE
Cleanup channel list

### PR DESCRIPTION
**Pull request by lionelvillard**
## Proposed Changes 

- Remove deprecated channels from the list to avoid confusion
- Change Support to Maintainer to better reflect Knative (or third-party) role.
- InMemory channel is "Stable" in a sense that it's compliant, maintained and does what it's supposed to do. 

/cc @pierDipi

### CODE REVIEW

File: docs/eventing/channels/channels-crds.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-# Available Channels
+---
+title: "Available Channels"
+#linkTitle: "Channels"
+weight: 40
+type: "docs"
+---
+
```
Comment (abrennan89) [inline docs/eventing/channels/channels-crds.md:7]: It was fine as it was, this block was the old frontmatter for Hugo and isn't needed now 🙂

File: docs/eventing/channels/channels-crds.md [hunk 2]
```diff
@@ -13,13 +19,25 @@ procedure:
 
 This is a non-exhaustive list of the available Channels for Knative Eventing.
 
-!!! note
-    Inclusion in this list is not an endorsement, nor does it imply any level of support.
+Notes:
+
+* Inclusion in this list is not an endorsement, nor does it imply any level of
+  support.
```
Comment (abrennan89) [inline docs/eventing/channels/channels-crds.md:25]: same re notes, the `!!! note` format is correct, see https://github.com/knative/docs/blob/main/contribute-to-docs/getting-started/formatting.md#admonitions

File: docs/eventing/channels/channels-crds.md [hunk 3]
```diff
@@ -13,13 +19,25 @@ procedure:
...
 [NatssChannel](https://github.com/knative-sandbox/eventing-natss/tree/{{ branch }}/config/README.md) | Proof of Concept | None | Channels are backed by [NATS Streaming](https://github.com/nats-io/nats-streaming-server#configuring).
 [KafkaChannel - Consolidated](https://github.com/knative-sandbox/eventing-kafka/tree/{{ branch }}/pkg/channel/consolidated/README.md) | Deprecated | None | Channels are backed by [Apache Kafka](http://kafka.apache.org/) topics. The original Knative KafkaChannel implementation which utilizes a single combined Kafka Producer / Consumer deployment.
 [KafkaChannel - Distributed](https://github.com/knative-sandbox/eventing-kafka/tree/{{ branch }}/pkg/channel/distributed/README.md) | Deprecated | None | Channels are backed by [Apache Kafka](http://kafka.apache.org/) topics. An alternate KafkaChannel implementation, contributed by SAP's [Kyma](https://kyma-project.io/) project, which provides a more granular deployment of Producers / Consumers.
+=======
+>>>>>>> 1314d755 (Cleanup channel list)
```
Comment (abrennan89) [inline docs/eventing/channels/channels-crds.md:43]: some leftover bits from a rebase need to be removed 😉

**Review by abrennan89 — CHANGES_REQUESTED** [review]

**Comment by pierDipi** [comment]
+1 for the idea!

**Review by abrennan89 — APPROVED** [review]
/lgtm

**Pull request merged**